runtime/verbose/verbose.c

```diff
@@ -780,15 +781,15 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved) {
 			vm->setVerboseState = &setVerboseState;
 			omrthread_monitor_init( &vm->verboseStateMutex, 0 );
 			if( vm->verboseStateMutex == NULL ) {
-				loadInfo->fatalErrorStr = "cannot allocate verboseStateMutex in verbose init";
+				vm->internalVMFunctions->setErrorJ9dll(PORTLIB, loadInfo, "cannot allocate verboseStateMutex in verbose init", FALSE);
 				goto _error;
 			}
 
 			/* Note - verboseStruct not needed for modron verbose gc */
 			initializeVerboseFunctionTable(vm);
 
 			if (!checkOptsAndInitVerbosegclog(vm, vm->vmArgsArray)) {
-				loadInfo->fatalErrorStr =  (char*)j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG|J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_VERB_FAILED_TO_INITIALIZE,"Failed to initialize.");
+				vm->internalVMFunctions->setErrorJ9dll(PORTLIB, loadInfo, j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG|J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_VERB_FAILED_TO_INITIALIZE,"Failed to initialize."), FALSE);
```

Comment:
Is the problem that I need to pass a local into `parseVerboseArgumentList` instead? If so, would we need to call `setErrorJ9dll` both inside and after the if statement because of the `goto`?
```cpp
const char *errorString = loadInfo->fatalErrorStr;
if (!parseVerboseArgumentList(vm, loadInfo, &errorString)) {
    vm->internalVMFunctions->setErrorJ9dll(PORTLIB, loadInfo, errorString, FALSE);
    goto _error;
}
vm->internalVMFunctions->setErrorJ9dll(PORTLIB, loadInfo, errorString, FALSE);
```